esdt-safe/src/lib.rs

```diff
@@ -280,14 +280,70 @@ pub trait EsdtSafe:
                 *burned += &actual_bridged_amount;
             });
         }
-        self.create_transaction_event(
+        TransactionDetails {
             batch_id,
             tx_nonce,
             payment_token,
             actual_bridged_amount,
             required_fee,
-            tx.to,
-            tx.from,
+            to_address: to.as_managed_buffer().clone(),
+            caller_address: caller.as_managed_buffer().clone(),
+        }
+    }
+
+    // endpoints
+
+    /// Create an MultiversX -> Ethereum transaction. Only fungible tokens are accepted.
+    ///
+    /// Every transfer will have a part of the tokens subtracted as fees.
+    /// The fee amount depends on the global eth_tx_gas_limit
+    /// and the current GWEI price, respective to the bridged token
+    ///
+    /// fee_amount = price_per_gas_unit * eth_tx_gas_limit
+    #[payable("*")]
+    #[endpoint(createTransaction)]
+    fn create_transaction(&self, to: EthAddress<Self::Api>) {
+        let transaction_details = self.create_transaction_common(to);
+        let token_nonce = self.call_value().single_esdt().token_nonce;
```

Comment:
I would move the `token_nonce == 0` requirement to `self.create_transaction_common` to remove duplicate code.
Also it's better to find verify the requirement and after that execute code.